futures-util/src/stream/stream/mod.rs

```diff
@@ -315,7 +315,9 @@ pub trait StreamExt: Stream {
     ///
     /// Note that this function consumes the stream passed into it and returns a
     /// wrapped version of it, similar to the existing `filter` methods in the
-    /// standard library.
+    /// standard library. Because of current HRTB limitations, reference to item
```

Comment:
I don't think this should be added-- I think it will confuse users more than help them.